### PR TITLE
Import Export

### PR DESCRIPTION
**Pull request by ScriptHubs**

### CODE REVIEW

File: app/Imports/ImportCustomersCustomer.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class ImportCustomersCustomer implements ToModel, WithHeadingRow
+{
+    /**
+    * @param array $row
+    *
+    * @return \Illuminate\Database\Eloquent\Model|null
+    */
+    public function model(array $row)
+    {
+        return new Customers([
```
Comment (xyberpastoril) [inline app/Imports/ImportCustomersCustomer.php:18]: image
Temporarily change catch error message to `$e->getMessage()` to see this error.

File: app/Imports/ImportBankAccount.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class ImportBankAccount implements ToModel, WithHeadingRow
+{
+    /**
+    * @param array $row
+    *
+    * @return \Illuminate\Database\Eloquent\Model|null
+    */
+    public function model(array $row)
+    {
+        $coa = new ChartOfAccounts();
```
Comment (xyberpastoril) [inline app/Imports/ImportBankAccount.php:19]: image

File: app/Imports/ImportBankTransfer.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+use Maatwebsite\Excel\Concerns\WithValidation;
+
+class ImportBankTransfer implements ToModel, WithHeadingRow, WithValidation
+{
+    /**
+    * @param array $row
+    *
+    * @return \Illuminate\Database\Eloquent\Model|null
+    */
+    public function model(array $row)
```
Comment (xyberpastoril) [inline app/Imports/ImportBankTransfer.php:24]: image

**Review by xyberpastoril — CHANGES_REQUESTED** [review]
Kindly give these issues a check. I may do another review sometime later to check for potential pitfalls.

File: app/Imports/ImportCustomersCustomer.php (hunk 1 above)
Comment (ScriptHubs) [reply app/Imports/ImportCustomersCustomer.php:18]: This kind of error happens when user has no input in a row on import file but is not `nullable` in migrations.  Enhanced the validation. Will implement on other import/export if approved.

File: app/Http/Controllers/BankAccountsController.php [hunk 4]
```diff
@@ -17,16 +20,18 @@ class BankAccountsController extends Controller
...
+
+
         // $coa_last_record = BankAccounts::orderBy('created_at', 'desc')->first();
         // if (empty($coa_last_record)) {
         //     $coa_number = 1030;
         // } else {
         //     $coa_number = $coa_last_record->chartOfAccount->chart_of_account_no + 1;
         // }
-        return view('banking.accounts.index', compact('bank_accounts','coa_number'));
+        return view('banking.accounts.index', compact('bank_accounts'));
```
Comment (xyberpastoril) [inline app/Http/Controllers/BankAccountsController.php:34]: Good catch!

**Review by xyberpastoril — APPROVED** [review]

**Pull request merged**